user-guide/Advanced_Functionality/DataMiner_Systems/BPA_tests/BPA_SRM_Function_Resources_Consistency.md

```diff
@@ -0,0 +1,80 @@
+
+## Metadata
+
+- Name: SRM Function Resources Consistency
+- Description: Detects any inconsistencies between the resource objects and the [Generic DVE table] entries on the main element as well as duplicate names.
+- Author: Skyline Communications
+- Default schedule: Daily
+
+## Results
+
+### Success
+
+`Test succeeded: No inconsistent function resource configurations detected in the system.`
+
+There are no inconsistencies detected in the system.
+
+### Error
+
+- `[Generic DVE Table] entry {PK} is configured with resource ID '{ResourceID}', which was not found in Resource Manager.`
+- `[Generic DVE Table] entry {PK} is configured with a blank resource ID, so no matching resource could be found in the Resource Manager.`
+- `[Generic DVE Table] entry {PK} is configured with the element name - function name combination '{Name}.{Name}', which is already used as the element name of element '{DataMinerID}/{ElementID}'.`
+- `[Generic DVE Table] entries [{PKs}] are configured with a duplicate DVE Name '{Name}'.`
+- `The function name '{Name}' of resource '{Name}' ({GUID}) with [Generic DVE Table] primary key [{PK}] is already used by one or more other function resources on the same parent element, with [Generic DVE Table] primary key(s) [{PKs}].`
+- `Resource '{Name}' ({GUID}) has no primary key while it should have one.`
+- `Resource '{Name}' ({GUID}) with [Generic DVE Table] primary key '{PK}' could not be found in the [Generic DVE table].`
+- `The [Generic DVE Table] was empty, but the following resources refer to this element: Resource '{Name}' ({GUID}).`
+- `Could not find main element referenced by the following resources: {Names}. `
+- `The ElementID is not unique.`
+- `Resource '{Name}' ({GUID}):`
+    - `[Generic DVE Table] entry {PK} - [DVE Name] has value '{Name}', which is different from the linked resource function name '{Name}'.`
+    - `[Generic DVE Table] entry {PK} - [DVE Element] has the value '{Value}', which is different from the linked resource DVE Element '{DataMinerID}/{ElementID}'.`
+    - `[Generic DVE Table] entry {PK} - [DVE State] has the value 'Enabled', but [DVE Element] '{DataMinerID}/{ElementID}' was not found in the DataMiner System even though it should be.`
+    - `[Generic DVE Table] entry {PK} - [DVE State] has the value 'Disabled', but [DVE Element] '{DataMinerID}/{ElementID}' is present in the DataMiner System even though it shouldn't be.`
+    - `[Generic DVE Table] entry {PK} - [DVE function GUID] has the value '{GUID}', which is different from the linked resource function GUID '{GUID}'.`
+    - `[Generic DVE Table] entry {PK} - [DVE Link to Resource Manager] has the value '{Value}', which is different from the linked resource ID '{GUID}'.`
+    - `[Generic DVE Table] entry {PK} - The active function files for the protocol were switched, but [DVE function GUID] still matches the GUID of the function in a previous active function file('{GUID}') instead of the current active function file.`
```

Comment:
```suggestion
    - `[Generic DVE Table] entry {PK} - The active function files for the protocol were switched, but [DVE function GUID] still matches the GUID of the function in a previous active function file ('{GUID}') instead of the current active function file.`
```